### PR TITLE
Fix jax.numpy.linalg.inv with shape polymorphism

### LINKED ISSUE #22489: jax.numpy.digitize doesn't work with shape polymorphism
### Description

Error when tracing with shape polymorphism in jax.numpy.digitize
```
  File ".../lib/python3.12/site-packages/jax/_src/numpy/lax_numpy.py", line 7548, in searchsorted
    dtype = int32 if len(a) <= np.iinfo(np.int32).max else int64
                     ^^^^^^
TypeError: '_DimExpr' object cannot be interpreted as an integer
```

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:    0.4.30
jaxlib: 0.4.30
numpy:  2.0.0
```

### PR DESCRIPTION
**Pull request by tchatow**
Fixes https://github.com/google/jax/issues/22489#issuecomment-2338868969

### CODE REVIEW

**Review by gnecula — APPROVED** [review]
Thank you!

File: tests/shape_poly_test.py [hunk 1]
```diff
@@ -1097,6 +1097,10 @@ def f1(x, y):
     exp = jax.export.export(jax.jit(f1))(*args_specs)
     self.assertEqual(exp.in_avals[0], exp.in_avals[1])
 
+  def test_linalg_inv_bug_22489(self):
```
Comment (gnecula) [inline tests/shape_poly_test.py:1100]: Please move this test to be part of_POLY_SHAPE_TEST_HARNESSES, e.g., search for how "linspace" is tested. That way the test follows the same pattern as others and it is more complete (it actually runs the code).

**Review by gnecula — CHANGES_REQUESTED** [review]

**Comment by dfm** [comment]
You should be able to still add the test by adding:

```python
override_jax_config_flags={"jax_export_ignore_forward_compatibility": True}
```

to the test harness like we do for the LU test just a couple below the one you've added. I'm pretty sure that'll be enough!

**Comment by tchatow** [comment]
Got it. Unfortunately I couldn't figure out a non-trivial random test case, since neither of the ways I can think of to check for invertibility - `jnp.linalg.cond` or `jnp.linalg.matrix_rank` work with shape polymorphism.

**Review by gnecula — APPROVED** [review]

**Pull request merged**